Test infra: login rate-limit (60/min) makes full pytest suite flaky

## Problem
After the #90 isolation fix, the full backend suite is order-independent, but it's now flaky due to the **login endpoint rate limit (60 requests / 1 minute)**. Many tests create+login a unique user; when enough run within a minute, later tests' logins get rejected:

```
AssertionError: Login failed for <user>: {"error":"Rate limit exceeded: 60 per 1 minute"}
```

## Evidence (2026-05-24)
- Running the full suite, ~8 `tests/v3/test_pipelines.py` tests (and some `test_account_endpoints.py`) fail with the login-429 above.
- The SAME files pass cleanly **in isolation**: `pytest tests/v3/test_pipelines.py` → **50 passed, 0 failed**.
- So it's not a code bug or fixture-isolation bug (those were fixed in #90) — it's cumulative login pressure tripping the rate limiter. Adding tests (e.g. #128's 28 new tests) shifts *which* tests trip it, making the failure set non-deterministic.

## Impact
- Full-suite green/red is unreliable; CI and local `pytest tests/` give different failure sets per run.
- Forces per-file isolation runs to verify correctness (as done when merging #128).

## Suggested fix (pick one)
1. **Bypass/raise the rate limiter under test** — detect a test env (e.g. `PYTEST_CURRENT_TEST` / a `TESTING=1` env) and skip slowapi limiting on `/v3/auth/login`, or set a very high limit.
2. **Session-scoped token reuse** — a conftest fixture that logs in once per user and caches the token, instead of re-logging-in per test.
3. Mark login-heavy tests to share a token fixture.

Option 1 is simplest and most robust. Relates to #90 (test isolation).

--- a/app/lib/rate_limit.py
+++ b/app/lib/rate_limit.py
@@ -9,9 +9,17 @@ Override per-route in the route decorator.
 """
 from __future__ import annotations
 
+import os
+
 from fastapi import Request
 from slowapi import Limiter
 from slowapi.util import get_remote_address
+
+# Disable limiting under test: the suite makes hundreds of requests from a
+# single client IP (one bucket), which trips the 60/minute default and causes
+# flaky `429`s unrelated to the code under test. conftest sets DISABLE_RATE_LIMIT=1
+# before importing the app. (Does not affect prod, where the env var is unset.)
+_RATE_LIMIT_ENABLED = os.getenv("DISABLE_RATE_LIMIT") != "1"
 
 
 def _key_func(request: Request) -> str:
@@ -31,4 +39,5 @@ limiter = Limiter(
     default_limits=["60/minute"],
     storage_uri="memory://",
     headers_enabled=True,
+    enabled=_RATE_LIMIT_ENABLED,
 )

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -20,3 +20,13 @@ import psycopg2.extras  # noqa: F401
 import os as _os
 _os.environ.setdefault("TEMPORAL_HOST", "localhost")
 _os.environ.setdefault("IS_USE_TEMPORAL", "false")
+# Disable the slowapi rate limiter under test (#130): the suite makes hundreds
+# of requests from one client IP, tripping the 60/min default and causing flaky
+# 429s ("Rate limit exceeded: 60 per 1 minute") on login-heavy tests. Must be
+# set before app import so the Limiter is constructed with enabled=False.
+_os.environ.setdefault("DISABLE_RATE_LIMIT", "1")
+# Bypass the run admission gate under test (#130): GLOBAL_MAX_CONCURRENT is 2,
+# so any stuck/concurrent runs in the shared dev DB make `check_admission` 429
+# every /v3/agent/message in the suite. Tests shouldn't depend on global run
+# capacity. Read at admission_gate import time.
+_os.environ.setdefault("GATE_ENABLED", "false")

